hicommonwealth/commonwealth
# User-Onboarding: Random username generation + Unique usernames

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/6751

## Description of Changes
- Added random username generation
- Added username uniqueness checks

## "How We Fixed It"


## Test Plan

- Enable `FLAG_USER_ONBOARDING_ENABLED` in `.env`.
- Add `type='create-account'` to `<AuthModal/>` instance in `SubLayoutHeader.tsx` or apply this diff
```diff
diff --git a/packages/commonwealth/client/scripts/views/components/SublayoutHeader/SublayoutHeader.tsx b/packages/commonwealth/client/scripts/views/components/SublayoutHeader/SublayoutHeader.tsx
index 4564d26340..c58647b700 100644
--- a/packages/commonwealth/client/scripts/views/components/SublayoutHeader/SublayoutHeader.tsx
+++ b/packages/commonwealth/client/scripts/views/components/SublayoutHeader/SublayoutHeader.tsx
@@ -59,6 +59,7 @@ export const SublayoutHeader = ({
         open={isFeedbackModalOpen}
       />
       <AuthModal
+        type='create-account'
         onClose={() => setIsAuthModalOpen(false)}
         isOpen={isAuthModalOpen}
       />

```
open the modal and create a new web3 wallet account, verify you see the `Welcome to Common` modal after a new account is created.
- Verify that you can generate a random username with the `Generate random username` button
- Verify that the generated or user entered username is validated as unique, if its not unique there will be a `Username already exists` error below the username input

## Deployment Plan
N/A

## Other Considerations
N/A
